test_app/main/test_app_main.c

```diff
@@ -11,6 +11,8 @@
 void app_main(void)
 {
     UNITY_BEGIN();
-    unity_run_menu();
+    //unity_run_tests_by_tag("[libsodium]", false);
+    unity_run_all_tests();
+    //unity_run_menu();
```

Comment:
Are the commented out lines needed? If no, can they be removed?

(I wonder if the reason for keeping them is that you'd like to have different behavior when running tests in CI and when running them manually. In that case we could add some Kconfig option to the test app ("Run all tests automatically") and enable it only in CI builds.)